File: internal/content/testdata/bench/go/next-lexicographic-permutation.go
func next_permutation(nums []int) []int {
	n := len(nums)
	pivot := -1
	for i := n - 2; i >= 0; i-- {
		if nums[i] < nums[i+1] {
			pivot = i
			break
		}
	}
	if pivot == -1 {
		for i, j := 0, n-1; i < j; i, j = i+1, j-1 {
			nums[i], nums[j] = nums[j], nums[i]
		}
		return nums
	}
	for r := n - 1; r > pivot; r-- {
		if nums[r] > nums[pivot] {
			nums[pivot], nums[r] = nums[r], nums[pivot]
			break
		}
	}
	for i, j := pivot+1, n-1; i < j; i, j = i+1, j-1 {
		nums[i], nums[j] = nums[j], nums[i]
	}
	return nums
}

File: internal/content/testdata/bench/javascript/next-lexicographic-permutation.js
function next_permutation(nums) {
  const n = nums.length;
  let pivot = -1;
  for (let i = n - 2; i >= 0; i--) {
    if (nums[i] < nums[i + 1]) {
      pivot = i;
      break;
    }
  }
  if (pivot === -1) {
    nums.reverse();
    return nums;
  }
  for (let r = n - 1; r > pivot; r--) {
    if (nums[r] > nums[pivot]) {
      [nums[pivot], nums[r]] = [nums[r], nums[pivot]];
      break;
    }
  }
  const suffix = nums.splice(pivot + 1).reverse();
  nums.push(...suffix);
  return nums;
}
